charms/worker/k8s/src/certificates.py

```diff
@@ -335,26 +337,90 @@ def _populate_bootstrap_certificates(self, bootstrap_config: BootstrapConfig) ->
         bootstrap_config.kubelet_cert = str(certificate.certificate)
         bootstrap_config.kubelet_key = str(key)
 
-    def _get_certificates_provider(self) -> Optional[str]:
-        """Get the certificates provider.
+    def get_provider_name(self) -> str:
+        """Get the certificates provider name.
 
         Returns:
             str: The certificates provider.
         """
-        if self._charm.is_control_plane:
-            return BOOTSTRAP_CERTIFICATES.get(self._charm)
-
-        # NOTE: This operation is safe because we're validating the provider during the
-        # certificate configuration in the `configure_certificates` method.
+        app = self._charm.is_control_plane and self._charm.app or None
         relation = self.model.get_relation(CLUSTER_RELATION)
-        if not relation:
-            return None
+        provider = relation and relation.data[app or relation.app].get(CLUSTER_CERTIFICATES_KEY)
 
-        if not (provider := relation.data[relation.app].get(CLUSTER_CERTIFICATES_KEY)):
-            log.info("Waiting for certificates provider")
-            return None
+        if not provider:
+            # Note(AKD): This could be an upgrade scenario where the provider is unset.
+            # if this node is online, we know that a certificates provider is already set.
+            try:
+                self._charm.api_manager.get_node_status().metadata
+                provider = "self-signed" if pki.check_ca_key() else "external"
+            except (K8sdConnectionError, InvalidResponseError) as e:
+                log.error("Failed to get node status: %s", e)
```

Comment:
Should we raise here and prevent the charm to keep going with an `None` value? I see the logic blocks while validating the provider, but I'd prefer a different approach, as the status `Invalid certificate issuer: ''` could be misleading.

Reply:
I think i went down that chain at first, but there's this chicken/egg issue with the `K8sCertificates` class is used outside the reconciler, so raising here wouldn't be good..

```
K8sCharm.__init__(...)
  K8sCertificates.__init__(...)
    K8sCertificates._init_certificates_provider(...)
       K8sCertificates.get_provider_name(...)
... HERE ...
```

Reply:
also, this error you mentioned wouldn't show up unless the config option itself is set to `""`

```python
        if not provider and self._charm.is_control_plane:
            provider = BOOTSTRAP_CERTIFICATES.get(self._charm)
            self._validate_provider(provider)
```

I should deploy the charm with these changes with `juju deploy k8s --config bootstrap-certificates=""` and see how this is handled.  We'll likely end up in an error state.